src/MongoDB.Analyzer/Core/Builders/BuildersResolveVariables.cs

```diff
@@ -0,0 +1,432 @@
+                var leftHandSideIdentifier = assignmentExpression.Left.TrimParenthesis() as IdentifierNameSyntax;
+                canEvaluate = variableValues.TryGetValueSafe(leftHandSideIdentifier.Identifier.ValueText, out var leftVariableValue);
+
+                if (leftVariableValue != null)
+                {
+                    syntaxNodeToVariableValueMap ??= new Dictionary<SyntaxNode, VariableValue>();
+                    syntaxNodeToVariableValueMap.Add(leftHandSideIdentifier, leftVariableValue);
+                }
+            }
+        }
+
+        return new EvaluationResult(canEvaluate, syntaxNodeToVariableValueMap);
+    }
+
+    private static ExpressionAnalysisContext GenerateNewExpressionAnalysisContext(
+        SyntaxNode syntaxNode,
+        Dictionary<SyntaxNode, VariableValue> builderNodeToVariableValue)
+    {
+        var argumentTypeName = builderNodeToVariableValue.Values.First().ExpressionAnalysisContext.Node.ArgumentTypeName;
+        var rewrittenSyntaxNode = syntaxNode.ReplaceNodes(builderNodeToVariableValue.Keys, (n, _) => builderNodeToVariableValue[n].ExpressionAnalysisContext.Node.RewrittenExpression);
+
+        if (IsCompoundExpression(syntaxNode))
+        {
+            var rewrittenAssignmentExpression = rewrittenSyntaxNode as AssignmentExpressionSyntax;
+            var binaryExpressionSyntaxKind = GenerateBinaryExpressionSyntaxKind(syntaxNode);
+            rewrittenSyntaxNode = SyntaxFactory.BinaryExpression(binaryExpressionSyntaxKind, rewrittenAssignmentExpression.Left, rewrittenAssignmentExpression.Right);
+        }
+
+        var rewrittenBuildersExpression = rewrittenSyntaxNode switch
+        {
+            AssignmentExpressionSyntax @assignmentExpression => @assignmentExpression.Right,
+            VariableDeclaratorSyntax @variableDeclarator => @variableDeclarator.Initializer.Value,
+            _ => rewrittenSyntaxNode
+        };
+
+        var (finalBuildersExpression, finalConstantMapper) = ConstantsMapperComposer.Compose(rewrittenBuildersExpression, builderNodeToVariableValue.Values.Select(v => v.ExpressionAnalysisContext.Node.ConstantsRemapper));
+        return new ExpressionAnalysisContext(new ExpressionAnalysisNode(rewrittenBuildersExpression, argumentTypeName, finalBuildersExpression, finalConstantMapper));
+    }
+
+    private static bool ProcessDeclaration(VariableDeclaratorSyntax variableDeclarator, ContextInformation contextInformation, HashSet<string> dirtyVariables, bool displayDiagnostics)
+    {
+        var evaluationResult = PreEvaluate(variableDeclarator, contextInformation, null, displayDiagnostics);
+        var variableName = variableDeclarator.Identifier.ValueText;
+
+        if (evaluationResult.CanEvaluate)
+        {
+            var variableValue = evaluationResult.SyntaxNodeToVariableValueMap.Values.First();
+            var generateNewContext = evaluationResult.SyntaxNodeToVariableValueMap.Count() > 1;
```

Comment:
add evaluationResult.HasMultipleVariables or similar.

HasMultipleVariables  =>  SyntaxNodeToVariableValueMap?.Count > 1;

Use Count instead of Count() (Linq method).

Reply:
Done